internal/clients/http_client/http_client.go

```diff
@@ -64,6 +64,19 @@ func NewBasicHTTPClient(baseURL string, timeout time.Duration) *defaultHTTPClien
 	}
 }
 
+func NewAuthenticatedHTTPClient(baseURL string, timeout time.Duration, authMethod auth.AuthMethod) *defaultHTTPClient {
```

Comment:
This one does not seem to be used

Reply:
This one still seems to not be used.

Reply:
Aaah yes, I missed this in my last clean up, thanks for the (second) nudge 🙏 

I've gotten too used to some of the typescript helpers that yell at me when something is unused 🙈